recipes/libatomic_ops/all/conanfile.py

```diff
@@ -23,88 +27,78 @@ class Libatomic_opsConan(ConanFile):
-        return self._autotools
-
-    @contextlib.contextmanager
-    def _build_context(self):
-        if self.settings.compiler == "Visual Studio":
-            with tools.vcvars(self):
-                with tools.environment_append({"CC": "cl -nologo", "CXX": "cl -nologo",
-                                               "LD": "link"}):
-                    yield
-        else:
-            yield
-
-    @property
-    def _user_info_build(self):
-        return getattr(self, "user_info_build", None) or self.deps_user_info
+        get(self, **self.conan_data["sources"][self.version],
+                  destination=self.source_folder, strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        for option, _ in self._cmake_options_defaults:
+            tc.variables["enable_{}".format(option)] = self.options.get_safe(option)
+        tc.variables["install_headers"] = True
+        tc.variables["build_tests"] = False
+        tc.generate()
+        tc = CMakeDeps(self)
+        tc.generate()
 
     def build(self):
-        shutil.copy(self._user_info_build["gnu-config"].CONFIG_SUB,
-                    os.path.join(self._source_subfolder, "config.sub"))
-        shutil.copy(self._user_info_build["gnu-config"].CONFIG_GUESS,
-                    os.path.join(self._source_subfolder, "config.guess"))
-        with self._build_context():
-            autotools = self._configure_autotools()
-            autotools.make()
+        apply_conandata_patches(self)
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
 
     def package(self):
         self.copy("COPYING", src=self._source_subfolder, dst="licenses")
-
-        with self._build_context():
-            autotools = self._configure_autotools()
-            autotools.install()
-
-        tools.remove_files_by_mask(os.path.join(self.package_folder, "lib"), "*.la")
-        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
-        tools.rmdir(os.path.join(self.package_folder, "share"))
-        if self.settings.compiler == "Visual Studio" and self.options.shared:
-            for lib in ["atomic_ops_gpl", "atomic_ops"]:
-                tools.rename(os.path.join(self.package_folder, "lib", "{}.dll.lib".format(lib)),
-                             os.path.join(self.package_folder, "lib", "{}.lib".format(lib)))
+        cmake = CMake(self)
+        cmake.install()
+        rmdir(self, os.path.join(self.package_folder, "share"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
 
     def package_info(self):
-        self.cpp_info.names["pkg_config"] = "atomic_ops"
-        self.cpp_info.libs = ["atomic_ops_gpl", "atomic_ops"]
+        self.cpp_info.set_property("cmake_file_name", "Atomic_ops")
+        self.cpp_info.set_property("cmake_target_name", "Atomic_ops::Atomic_ops")
```

Comment:
Is there such target upstream?

Reply:
Yes, https://github.com/ivmai/libatomic_ops/blob/master/CMakeLists.txt#L340
https://github.com/ivmai/libatomic_ops/blob/master/CMakeLists.txt#L143

Reply:
No, this one is a component in package_info(), and it's name is `Atomic_ops::atomic_ops`

Reply:
see https://github.com/conan-io/conan-center-index/pull/13042#discussion_r975685600